configure.ac

```diff
@@ -87,6 +87,8 @@ AC_CHECK_PROGS([DOXYGEN], [doxygen])
 AM_CONDITIONAL(HAVE_DOXYGEN, test -n "${DOXYGEN}")
 AC_CHECK_PROGS([SPLINT], [splint])
 AM_CONDITIONAL(HAVE_SPLINT, test -n "${SPLINT}")
+AC_CHECK_PROGS([DPKG_ARCHITECTURE], [dpkg-architecture])
+AM_CONDITIONAL(HAVE_DPKG_ARCHITECTURE, test -n "${DPKG_ARCHITECTURE}")
```

Comment:
Thanks, looks good, except I'd like to keep the compile time checking
burden at absolute minimum (there surely are other suboptimal places,
but going forward, better not to add anything without proper
conditionalizing when not covering all use cases).

Hence wrap the two lines (and only that) with if test -n "${SPLINT}"
conditionalizing, please, and we are good (other two patches are OK).

Reply:
note from https://www.gnu.org/software/automake/manual/html_node/Usage-of-Conditionals.html is ' If AM_CONDITIONAL is run conditionally (e.g., in a shell if statement), then the result will confuse automake' hence this looking probably uglier than it should be.

Reply:
done.